### ISSUE TITLE
Annotated Invokes

### ISSUE THREAD

**Issue by fionera** [uber-go/fx#691]
Hi,

currently we need to call fx.Invoke with a custom Type with `fx.In` and the parameters. How about implementing support for fx.Annotated in fx.Invoke calls?

**Comment by abhinav** [comment]
Hey, we considered that but the problem is that constructors usually have
multiple parameters. A `name` or `group` specified with `fx.Annotated` would
apply to all parameters, which is not what you usually want.

For example, if you have,

    func New(ro *sql.DB, rw *sql.DB) Repository

There's no easy way to annotate the `name`s for individual parameters.
We're open to suggestions that address this concern.

**Comment by abhinav** [comment]
This feature is now available in [v1.15.0](https://github.com/uber-go/fx/releases/tag/v1.15.0) in the following form:

```
fx.Invoke(
  fx.Annotate(
    func(foo Thing) {
        // ...
    },
    fx.ParamTags(`name:"a"`),
  ),
)
```

Which is roughly equivalent to,

```
type invokeParams struct {
    fx.In
    
    Foo Thing `name:"a"`
}

fx.Invoke(
    func(p invokeParams) {
        foo := p.Foo

        // ...
    },
)
```
